os/fs/smartfs/smartfs_smart.c

```diff
@@ -220,7 +220,13 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 #endif							/* CONFIG_SMARTFS_USE_SECTOR_BUFFER */
 
 	sf->entry.name = NULL;
-	ret = smartfs_finddirentry(fs, &sf->entry, relpath);
+	if (oflags == O_RDOK) {
+		/* If the file is being opened only for reading, we need not calculate the file's length */
+		ret = smartfs_finddirentry(fs, &sf->entry, relpath, false);
```

Comment:
you divided smartfs_finddirentry but it still mixed.
You should consider when datalen is necessary.
In my opinion, we should reveal datalen with below condition 
1. Entry type is File
2. oflags is not read only.

Reply:
I have separated function wise, since most functions do not need finddirentry to calculate the data length.